test/AngularSpec.js

```diff
@@ -1671,6 +1671,120 @@ describe('angular', function() {
+
+      angular.bootstrap(element);
+      angular.shutdown(element);
+      expect(element.text()).toBe('Hello World');
+    });
+
+    it('should remove all the data from all the elements', function() {
+      var element = jqLite('<div><span>Bootstrap me!</span></div>');
+
+      angular.bootstrap(element);
+      element.find('span').data('foo', 'bar');
+      angular.shutdown(element);
+      expect(element.find('span').data('foo')).toBeUndefined();
+    });
+
+    describe('setTimeout and setInterval shutdown', function() {
+      var setTimeoutSpy, setInternalSpy;
+      beforeEach(inject(function($window) {
+        spyOn($window, 'setInterval').and.returnValue(42);
+        spyOn($window, 'clearInterval');
+        spyOn($window, 'setTimeout');
+      }));
+
+      it('should stop calling the window setTimeout', inject(function($window) {
+        var element = jqLite('<div><span>Bootstrap me!</span></div>'),
+          injector = angular.bootstrap(element),
+          $timeout = injector.get('$timeout');
+
+        $timeout(function() {});
+        expect($window.setTimeout).toHaveBeenCalledOnce();
+        angular.shutdown(element);
+        $timeout(function() {});
+        expect($window.setTimeout).toHaveBeenCalledOnce();
+      }));
+
+      it('should cancel all active interval calls', inject(function($window) {
+        var element = jqLite('<div><span>Bootstrap me!</span></div>'),
+          injector = angular.bootstrap(element),
+          $interval = injector.get('$interval');
+
+        $interval(function() {}, 100);
+        expect($window.setInterval).toHaveBeenCalledOnce();
+        expect($window.clearInterval).not.toHaveBeenCalled();
+        angular.shutdown(element);
+        expect($window.clearInterval).toHaveBeenCalledOnce();
+        expect($window.clearInterval.calls.mostRecent().args[0]).toBe(42);
+      }));
+
+      it('should not cancel intervals that were already canceled', inject(function($window) {
+        var element = jqLite('<div><span>Bootstrap me!</span></div>'),
+          injector = angular.bootstrap(element),
+          $interval = injector.get('$interval'),
+          intervalPromise = $interval(function() {}, 100);
+
+        $interval.cancel(intervalPromise);
+        expect($window.setInterval).toHaveBeenCalledOnce();
+        expect($window.clearInterval).toHaveBeenCalledOnce();
+        angular.shutdown(element);
+        expect($window.setInterval).toHaveBeenCalledOnce();
+        expect($window.clearInterval).toHaveBeenCalledOnce();
+      }));
+
+      it('should not call the window setInterval once it shutdown', inject(function($window) {
+        var element = jqLite('<div><span>Bootstrap me!</span></div>'),
+          injector = angular.bootstrap(element),
+          $interval = injector.get('$interval');
+
+        angular.shutdown(element);
+        expect($window.setInterval).not.toHaveBeenCalled();
+        expect($window.clearInterval).not.toHaveBeenCalled();
+      }));
```

Comment:
I assume the intention in this test was to call `$interval()` and verify that `$window.setInterval` hasn't been called, right?